File: src/java/magma/api/io/Path_.java
package magma.api.io;

import magma.api.collect.Set_;
import magma.api.collect.Stream;
import magma.api.option.Option;
import magma.api.result.Result;

public interface Path_ {
    Option<IOError> createDirectoriesSafe();

    Result<Set_<Path_>, IOError> walk();

    Option<IOError> writeString(String output);

    boolean isExists();

    Stream<String> stream();

    Path_ resolve(String segment);

    Path_ relativize(Path_ child);

    boolean isRegularFile();

    String asString();

    Option<Path_> findParent();

    Path_ last();

    Result<String, IOError> readString();
}

File: src/node/magma/api/io/Path.ts
/*[
	Actual, 
	Characters, 
	Collector, 
	Console, 
	EmptyHead, 
	Files, 
	FlatMapHead, 
	Head, 
	HeadedQuery, 
	IOError, 
	Joiner, 
	List, 
	ListCollector, 
	Lists, 
	MapHead, 
	Namespace, 
	Path, 
	Queries, 
	Query, 
	RangeHead, 
	SingleHead, 
	Strings, 
	ZipHead
]*/
import { IOError } from "../../../magma/api/io/IOError";
import { Option } from "../../../magma/api/option/Option";
import { Result } from "../../../magma/api/result/Result";
import { List } from "../../../magma/api/collect/list/List";
import { Query } from "../../../magma/api/collect/Query";
export interface Path {
	asString(): string;
	writeString(output: string): Option<IOError>;
	readString(): Result<string, IOError>;
	walk(): Result<List<Path>, IOError>;
	findFileName(): string;
	endsWith(suffix: string): boolean;
	relativize(source: Path): Path;
	getParent(): Path;
	query(): Query<string>;
	resolveChildSegments(children: List<string>): Path;
	resolveChild(name: string): Path;
	exists(): boolean;
	createDirectories(): Option<IOError>;
}
